src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -660,6 +659,82 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
         }
     }
 
+    /// Check fn in const context with supplied feature gates
+    fn check_fn_in_const_ctx(&mut self, def_id: DefId) -> () {
+        debug!("check_fn_in_const_ctx: {:?}", def_id);
+        // Only in non-const functions it is perfectly fine to call any
+        // function, even ones whose constness is unstable. Everywhere else
+        // we need to check the appropriate feature gates.
+        if self.mode.requires_const_checking() {
+            let unleash_miri = self
+                .tcx
+                .sess
+                .opts
+                .debugging_opts
+                .unleash_the_miri_inside_of_you;
+            debug!(
+                "check_fn_in_const_ctx: {:?} is const fn: {:?}",
+                def_id, self.tcx.is_const_fn(def_id),
+            );
+            debug!("is_raw_const_fn: {}", self.tcx.is_const_fn_raw(def_id));
+            debug!("is_unstable: {:?}", self.tcx.is_unstable_const_fn(def_id));
+            debug!("stability: {:?}", self.tcx.lookup_stability(def_id));
```

Comment:
@oli-obk you probably want to remove this.

Reply:
That was for debugging purposes I think, it will be removed when we are about to merge. Right? @oli-obk

Reply:
Yes